pkg/syncer/metadatasyncer.go

```diff
@@ -522,19 +522,19 @@ func pvcDeleted(obj interface{}, metadataSyncer *metadataSyncInformer) {
 		log.Errorf("PVCDeleted: Error getting Persistent Volume for pvc %s in namespace %s with err: %v", pvc.Name, pvc.Namespace, err)
 		return
 	}
-
-	// Verify if pv is a vsphere csi volume
-	if (pv.Spec.CSI == nil || pv.Spec.CSI.Driver != csitypes.Name) && (metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration) && pv.Spec.VsphereVolume == nil) {
-		log.Debugf("PVCDeleted: Not a vSphere CSI Volume or migrated vSphere Volume")
-		return
-	}
-
-	// Verify if pv is a vsphere volume and migration flag is set to false
-	if pv.Spec.VsphereVolume != nil && !metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration) {
-		log.Warnf("PVCDeleted: volume-migration feature switch is disabled. Cannot delete vSphere volume metadata for %s", pvc.Name)
-		return
+	migrationEnabled := metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration)
+	if migrationEnabled && pv.Spec.VsphereVolume != nil {
+		if !migratedToAnnotationExists(ctx, pvc.Annotations) {
+			log.Info("migration annotation is not present on the PVC. Skipping update")
```

Comment:
add "PVCDeleted" in the beginning of log message

Reply:
Shouldn't the log message with "Skipping delete" ?

Reply:
Info -> Debug. And also log the PVC.

Reply:
Same comment in other places too.

Reply:
fixed.